src/viewer.py

```diff
@@ -654,6 +656,7 @@ def on_set_initial_state(self, event):
 
     def on_trigger_transition(self, event):
         """Event: Change the current state of the server."""
+        # TODO This never works the first time it is executed.
```

Comment:
Hi @janEbert,
a possible reason for this issue might be the creation of the transition_pub publisher below.
- First, it is created from scratch for each call, which is not necessary
- Second, the initial creation overhead might cause the issues
Please move the creation of this publisher into the constructor.